server/channels/api4/channel.go

```diff
@@ -843,6 +845,18 @@ func getAllChannels(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func SanitizeChannels(c *Context, channels model.ChannelListWithTeamData) model.ChannelListWithTeamData {
+	if !c.App.SessionHasPermissionToAny(*c.AppContext.Session(), []*model.Permission{
+		model.PermissionSysconsoleReadComplianceDataRetentionPolicy,
```

Comment:
In lines 796-800, we check permissions `PermissionSysconsoleWriteUserManagementGroups` and `PermissionSysconsoleReadUserManagementChannels` and if the session has none of them we return a permission error. 

So if the session has only `PermissionSysconsoleReadComplianceDataRetentionPolicy`, the flow will never reach the new `SanitizeChannels` that takes into account this permission. 

We either need to add `PermissionSysconsoleReadComplianceDataRetentionPolicy` in the check in lines 796-800 if this is needed for proper functionality, or we need to remove the permission from here. 

if we add the  `PermissionSysconsoleReadComplianceDataRetentionPolicy` permission in the check in lines 796-800, we'll also need to add it in the `getPublicChannelsForTeam` function.

Reply:
@esarafianou Yeah. Looks like an assumption was made regarding `PermissionSysconsoleReadComplianceDataRetentionPolicy` having the other permissions. I added to `getAllChannels` and `searchAllChannels` but don't understand why its needed in `getPublicChannelsForTeam`, that's an unrelated API.

Reply:
I had a look and I don't understand that either: 
> why its needed in getPublicChannelsForTeam, that's an unrelated API.